src/Microsoft.AspNetCore.OData/Batch/ODataBatchReaderExtensions.cs

```diff
@@ -236,5 +268,103 @@ private static HttpContext CreateHttpContext(HttpContext originalContext)
 
             return context;
         }
+
+        /// <summary>
+        /// Extract preferences that can be inherited from the overall batch request to
+        /// an individual request.
+        /// </summary>
+        /// <param name="batchPreferences">The value of the Prefer header from the batch request</param>
+        /// <returns>comma-separated preferences that can be passed down to an individual request</returns>
+        private static string GetPreferencesToInheritFromBatch(string batchPreferences)
+        {
+            // do not inherit respond-async and continue-on-error (odata.continue-on-error in OData 4.0)
+            var preferencesToIgnore = new string[] { "respond-async", "continue-on-error", "odata.continue-on-error" };
+            var preferencesToInherit = batchPreferences.SplitPreferences()
+                .Where(value => 
+                !preferencesToIgnore.Any(
```

Comment:
May want to add indentation by one level to make clearer that it is part of the where clause...